Library/Homebrew/download_strategy.rb

```diff
@@ -66,9 +66,13 @@ def quiet?
     Context.current.quiet? || @quiet
   end
 
-  # Unpack {#cached_location} into the current working directory, and possibly
-  # chdir into the newly-unpacked directory.
-  # Unlike {Resource#stage}, this does not take a block.
+  # Unpack {#cached_location} into the current working directory.
+  #
+  # Additionally, if a block is given, the working directory was previously empty
+  # and a single directory is extracted from the archive, the block will be called
+  # with the working directory changed to that directory. Otherwise this method
+  # will return, or the block will be called, without changing the current working
+  # directory.
```

Comment:
```suggestion
  # Unpack {#cached_location} into the current working directory.
  #
  # If the working directory was previously empty and a single directory
  # is extracted from the archive, the staged directory is set to that single
  # directory. Otherwise, the staged directory is set to the current working directory. 
  #
  # If a block is given, the block is executed in the new staged directory.
  # If no block is given, the working directory is changed to the staged directory.
```

Here's a possible clarification